src/app.js

```diff
@@ -661,6 +685,43 @@ function resetDrawing() {
     }
 }
 
+// GRID FUNCTIONS
+function snapToGrid(x, y, gridSize = GRID_SIZE) {
+    const snappedX = Math.round(x / gridSize) * gridSize;
+    const snappedY = Math.round(y / gridSize) * gridSize;
```

Comment:
I think if we leave the floating-point realm the snapping will feel more precise:

```suggestion
    const snappedX = Math.floor((x + gridSize / 2) / gridSize) * gridSize;
    const snappedY = Math.floor((y + gridSize / 2) / gridSize) * gridSize; 
```